pkg/acsengine/munge.go

```diff
@@ -0,0 +1,194 @@
+package acsengine
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/Sirupsen/logrus"
+)
+
+const (
+	//Field names
+	customDataFieldName            = "customData"
+	dependsOnFieldName             = "dependsOn"
+	hardwareProfileFieldName       = "hardwareProfile"
+	imageReferenceFieldName        = "imageReference"
+	nameFieldName                  = "name"
+	osProfileFieldName             = "osProfile"
+	propertiesFieldName            = "properties"
+	resourcesFieldName             = "resources"
+	storageProfileFieldName        = "storageProfile"
+	typeFieldName                  = "type"
+	virtualMachineProfileFieldName = "virtualMachineProfile"
+	vmSizeFieldName                = "vmSize"
+
+	// ARM resource Types
+	nsgResourceType  = "Microsoft.Network/networkSecurityGroups"
+	vmResourceType   = "Microsoft.Compute/virtualMachines"
+	vmssResourceType = "Microsoft.Compute/virtualMachineScaleSets"
+)
+
+// NormalizeForVMSSScaling takes a template and removes elements that are unwanted in a VMSS scale up/down case
+func NormalizeForVMSSScaling(logger *logrus.Entry, templateMap map[string]interface{}) error {
+	if err := NormalizeMasterResourcesForScaling(logger, templateMap); err != nil {
+		return err
+	}
+
+	resources := templateMap[resourcesFieldName].([]interface{})
+	for _, resource := range resources {
+		resourceMap, ok := resource.(map[string]interface{})
+		if !ok {
+			logger.Warnf("Template improperly formatted")
+			continue
+		}
+
+		resourceType, ok := resourceMap[typeFieldName].(string)
+		if !ok || resourceType != vmssResourceType {
+			continue
+		}
+
+		resourceProperties, ok := resourceMap[propertiesFieldName].(map[string]interface{})
+		if !ok {
+			logger.Warnf("Template improperly formatted")
+			continue
+		}
+
+		virtualMachineProfile, ok := resourceProperties[virtualMachineProfileFieldName].(map[string]interface{})
+		if !ok {
+			logger.Warnf("Template improperly formatted")
+			continue
+		}
+
+		if !removeCustomData(logger, virtualMachineProfile) || !removeImageReference(logger, virtualMachineProfile) {
+			continue
+		}
+	}
+	return nil
+}
+
+// NormalizeForK8sVMASScalingUp takes a template and removes elements that are unwanted in a K8s VMAS scale up/down case
+func NormalizeForK8sVMASScalingUp(logger *logrus.Entry, templateMap map[string]interface{}) error {
```

Comment:
This was straight copied from RP. I think the 'S' is VMAS + Scaling up. I don't see an "extra" S.